diff -burN libffi-15.orig/powerpc/ppc-ffi_darwin.c libffi-15/powerpc/ppc-ffi_darwin.c
--- libffi-15.orig/powerpc/ppc-ffi_darwin.c	2008-08-30 01:28:23.000000000 +0200
+++ libffi-15/powerpc/ppc-ffi_darwin.c	2010-04-06 23:08:32.000000000 +0200
@@ -30,6 +30,7 @@
 
 #include <ffi.h>
 #include <ffi_common.h>
+#include <AvailabilityMacros.h>
 
 #include <stdbool.h>
 #include <stdio.h>
@@ -37,7 +38,7 @@
 #include <ppc-darwin.h>
 #include <architecture/ppc/mode_independent_asm.h>
 
-#if defined(POWERPC_DARWIN)
+#if defined(POWERPC_DARWIN) && (MAC_OS_X_VERSION_MIN_REQUIRED > MAC_OS_X_VERSION_10_4)
 #include <libkern/OSCacheControl.h>	// for sys_icache_invalidate()
 #endif
 
@@ -667,7 +668,7 @@
 		  +---------------------------------------+ xxx
 */
 
-#if !defined(POWERPC_DARWIN)
+#if defined(POWERPC_DARWIN) && (MAC_OS_X_VERSION_MIN_REQUIRED == MAC_OS_X_VERSION_10_4)
 
 #define MIN_LINE_SIZE 32
 
@@ -699,7 +700,7 @@
 	flush_icache(addr + size - 1);
 }
 
-#endif	// !defined(POWERPC_DARWIN)
+#endif	// defined(POWERPC_DARWIN) && (MAC_OS_X_VERSION_MIN_REQUIRED = MAC_OS_X_VERSION_10_4)
 
 ffi_status
 ffi_prep_closure(
@@ -747,7 +748,7 @@
 			closure->user_data = user_data;
 
 			// Flush the icache. Only necessary on Darwin.
-#if defined(POWERPC_DARWIN)
+#if defined(POWERPC_DARWIN) && (MAC_OS_X_VERSION_MIN_REQUIRED > MAC_OS_X_VERSION_10_4)
 			sys_icache_invalidate(closure->tramp, FFI_TRAMPOLINE_SIZE);
 #else
 			flush_range(closure->tramp, FFI_TRAMPOLINE_SIZE);
